### PR TITLE
move code for traffic graph to correct function; fix header if STATUS…

### PR DESCRIPTION
**Pull request by Skywalker-11**
…GRAPH_INTERVAL is undefined, reduce graph spacing

### CODE REVIEW

**Comment by stefaweb** [comment]
My PR #864 crossed yours.

If you like the idea, take it and I'll delete my PR.

**Comment by stefaweb** [comment]
Does `size AS total_size` needed?

> $graphgenerator->sqlQuery = '
>        SELECT
>        timestamp AS xaxis,
>        1 as total_mail,
>        virusinfected AS total_virus,
>        isspam AS total_spam,
>        size AS total_size
>       FROM
>        maillog
>       WHERE
>        1=1
>       AND
>        timestamp BETWEEN (NOW() - INTERVAL ' . $graphInterval . ' MINUTE) AND NOW()
>       ORDER BY
>        timestamp DESC
>      ';

File: mailscanner/conf.php.example [hunk 1]
```diff
@@ -301,5 +301,5 @@ define('RESET_LINK_EXPIRE', 1);
 // define('PWD_RESET_FROM_ADDRESS', 'support@yourdomain.com'); // Set is you wish reset email sent from alternative address.
 /* End Password Reset */
 
-//Interval for the traffic chart for admins (in minutes) default is 60
+// Interval for the traffic chart for admins (in mulitple of 60 minutes) default is 60
```
Comment (stefaweb) [inline mailscanner/conf.php.example]: multiple

**Comment by stefaweb** [comment]
What do think of this?

```
     SELECT
      timestamp AS xaxis,
      1 as total_mail,
      CASE 
      WHEN virusinfected > 0 THEN 1
      WHEN nameinfected > 0 THEN 1
      WHEN otherinfected > 0 THEN 1
      ELSE 0 END AS total_virus,
      CASE 
      WHEN isspam > 0 THEN 1
      WHEN ishighspam > 0 THEN 1
      WHEN issaspam > 0 THEN 1
      WHEN isrblspam > 0 THEN 1
      WHEN spamblacklisted > 0 THEN 1 
      ELSE 0 END AS total_spam
     FROM
      maillog
     WHERE
      1=1
     AND
      timestamp BETWEEN (NOW() - INTERVAL 120 MINUTE) AND NOW()
     ORDER BY
      timestamp DESC
```

File: mailscanner/style.css [hunk 2]
```diff
@@ -932,4 +932,6 @@ select.lang > option {
 #trafficgraph {
     width: auto;
     height: 180px;
+    margin-top: -4px;
+    margin-bottom: -13px;
```
Comment (stefaweb) [inline mailscanner/style.css]: Add `margin-right: -4px;`

**Comment by Skywalker-11** [comment]
This looks good
```
      WHEN virusinfected > 0 THEN 1
      WHEN nameinfected > 0 THEN 1
      WHEN otherinfected > 0 THEN 1
      ELSE 0 END AS total_virus,
``` 

But I think that this is unnecessary because as soon as they are in one of them the `isspam` is always true
```
      WHEN isspam > 0 THEN 1
      WHEN ishighspam > 0 THEN 1
      WHEN issaspam > 0 THEN 1
      WHEN isrblspam > 0 THEN 1
      WHEN spamblacklisted > 0 THEN 1 
      ELSE 0 END AS total_spam
```

File: mailscanner/style.css (hunk 2 above)
Comment (Skywalker-11) [inline mailscanner/style.css]: done

**Comment by stefaweb** [comment]
Ok for Query.

Question, is normal to have some columns NULL (not 0)?

capture d ecran 2017-06-16 a 17 24 10

I have not checked all concerned columns.

**Comment by Skywalker-11** [comment]
yes it is normal. At least it's the same here

**Comment by stefaweb** [comment]
Possible to have this kind of time legend?

legend

**Comment by Skywalker-11** [comment]
Theoretically yes but its not trivial.

**Comment by stefaweb** [comment]
And use the curent code but only print one value out of two?

**Comment by Skywalker-11** [comment]
The problem is that I only define what labels are there  but which label will be printed and how is Chart.js intern

**Comment by stefaweb** [comment]
Try this.

https://stackoverflow.com/questions/22064577/limit-labels-number-on-chart-js-line-chart

**Comment by stefaweb** [comment]
If you use  `ticks: { autoSkip: true, maxTicksLimit: 8 },` for xAxes, you got:

capture d ecran 2017-06-17 a 08 38 30

To evenly distribute ticks:

https://stackoverflow.com/questions/37250456/chart-js-evenly-distribute-ticks-when-using-maxtickslimit/37257056#37257056

**Comment by Skywalker-11** [comment]
thanks i will take a look at that

**Comment by stefaweb** [comment]
Need new style.

```
#trafficgraph {
    width: auto;
    height: 180px;
    margin-top: -4px;
    margin-right: -4px;
}
```

**Comment by Skywalker-11** [comment]
I changed the line graphs to use the time scales which fixes the gap in the labels. 
Still have the problem that the latest interval seems empty because the data point is aligned to the left of it. Going to look at it next week

**Pull request merged**